scripts/acknowledgements.js

```diff
@@ -0,0 +1,67 @@
+const fs = require('fs');
+
+const checker = require('license-checker');
+
+const packageInfo = fs.readFileSync('./package.json', { encoding: 'UTF-8' });
+
+// Licenses were taken from here:
+// https://tldrlegal.com/licenses/tags/open%20source
+const allowedLicenses = JSON.parse(fs.readFileSync(`${__dirname}/allowedLicenses.json`, { encoding: 'UTF-8' }));
+const dependencies = Object.keys(JSON.parse(packageInfo).dependencies);
+
+const acknowledgements = [];
+
+const isLicenseAllowed = (libInfo = {}) => {
+  const { licenses } = libInfo;
+
+  if (licenses instanceof Array) {
+    return licenses.every(license => allowedLicenses.indexOf(license) >= 0);
+  }
+
+  return allowedLicenses.indexOf(licenses) >= 0;
+};
+
+checker.init({
+  start: './',
+  unknown: true,
+}, (err, json) => {
+  if (err) {
+    console.error(err);
+  } else {
+    const strippedVersions = Object.keys(json).reduce((acc, curr) => ({
+      ...acc,
+      [curr.replace(/@\d.*$/gi, '')]: json[curr],
+    }), {});
+    dependencies.forEach((lib) => {
+      const libData = strippedVersions[lib];
+      if (!libData) {
+        throw new Error(`No data for library ${lib}`);
+      }
+      delete libData.publisher;
+      delete libData.email;
+      delete libData.url;
+      delete libData.path;
+      const libInfo = {
+        ...libData,
+        license: fs.readFileSync(strippedVersions[lib].licenseFile, { encoding: 'UTF-8' }),
+        name: lib,
+      };
```

Comment:
Do we need full license text? Probably we could store only licenses/repository/name fields.

Reply:
The thing is some of the licenses even though are open-source require users to distribute license text as well. So, we should have some sort of acknowledgements sections with licenses text of our dependencies in there.

This is the only reason for this.

Reply:
e.g.:

> The above copyright notice and this permission notice shall be\nincluded in all copies or substantial portions of the Software.